assets/src/edit-story/app/media/utils/createResource.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import getTypeFromMime from './getTypeFromMime';
+
+/**
+ * Resource object.
+ *
+ * TODO: Try to remove posterId (poster should be enough) and videoId (src)
+ * should be enough.
+ * TODO: We should just get the non-formatted numeric length. Formatting can
```

Comment:
See #714 for how we can get non-formatted numeric length

Reply:
Removed this TODO for now since it's mostly unrelated to this work. If we run into the need, we'll need to redo it more systematically.

Reply:
Ok. Now I understand what you were saying since I got the rebase conflict :). I caught up by branch on your changes and also added `length` for file upload API.